--- cinepaint/plug-ins/icc_examin/icc_examin/icc_gl.cpp.orig	2008-08-22 20:11:18.000000000 +0200
+++ cinepaint/plug-ins/icc_examin/icc_examin/icc_gl.cpp	2013-03-24 07:44:09.889892447 +0100
@@ -697,7 +697,7 @@
   if(font) delete font;
   if(ortho_font) delete ortho_font;
 
-  const char* font_name = "/usr/X11R6/lib/X11/fonts/truetype/FreeSans.ttf";
+  const char* font_name = "/usr/share/fonts/TTF/FreeSans.ttf";
 # if APPLE
   std::string f_n;
   if(getenv("RESOURCESPATH"))
@@ -714,7 +714,7 @@
 # endif
   if(!holeDateiModifikationsZeit(font_name)) {
     DBG_S( _("Could not open font in:") << font_name )
-    font_name = "/usr/X11R6/lib/X11/fonts/truetype/arial.ttf";
+    font_name = "/usr/share/fonts/TTF/arial.ttf";
     if(!holeDateiModifikationsZeit(font_name)) {
       DBG_S( _("Could not open font in:") << font_name )
 #ifdef  WIN32
